MinecraftClient/ExitCleanUp.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Runtime.InteropServices;
+
+namespace MinecraftClient
+{
+    /// <summary>
+    /// Perform clean up before quitting application
+    /// </summary>
+    /// <remarks>
+    /// Only ctrl+c/ctrl+break will be captured when running on mono
+    /// </remarks>
+    public static class ExitCleanUp
+    {
+        /// <summary>
+        /// Store codes to run before quitting
+        /// </summary>
+        private static List<Action> actions = new List<Action>();
+
+        static ExitCleanUp()
+        {
+            try
```

Comment:
You can also check `Program.isUsingMono` but catching `DllNotFoundException` is fine. See [ConsoleIcon.cs](https://github.com/ORelio/Minecraft-Console-Client/blob/master/MinecraftClient/WinAPI/ConsoleIcon.cs).